observatory_platform/templates/updated_example/specialised_telescope.py

```diff
@@ -0,0 +1,130 @@
+        super().__init__(dag_id, start_date, schedule_interval, extensions)
+        self.partition_table_id = partition_table_id
+
+    def create_airflow_DAG(self) -> DAG:
+        default_args = {
+            "owner": "airflow",
+            "start_date": self.start_date,
+            # 'on_failure_callback': on_failure_callback
+        }
+
+        with DAG(dag_id=self.dag_id, schedule_interval=self.schedule_interval,
+                 default_args=default_args, catchup=False, max_active_runs=1, doc_md=self.__doc__) as dag:
+            task1 = PythonOperator(task_id=self.TASK_ID_CREATE_RELEASE, provide_context=True,
+                                   python_callable=self.create_release_task)
+            task2 = ShortCircuitOperator(task_id=self.TASK_ID_DOWNLOAD, provide_context=True,
+                                         python_callable=self.download_task)
+            task3 = PythonOperator(task_id=self.TASK_ID_BQ_LOAD_PARTITION, provide_context=True,
+                                   python_callable=self.bq_load_partition_task)
+
+            task1 >> task2 >> task3
+        return dag
+
+    # not an abstractmethod, but can optionally be overwritten
+    def create_release(self, start_date: pendulum.Pendulum, end_date: pendulum.Pendulum, first_release: bool):
+        release = SpecialisedRelease(self.dag_id, start_date, end_date, self.extensions, first_release)
+        return release
+
+    def create_release_task(self, **kwargs):
+        """ Create a release instance and update the xcom value with the last start date.
+
+        :param kwargs: The context passed from the PythonOperator.
+        :return: None.
+        """
+        ti: TaskInstance = kwargs['ti']
+
+        first_release = False
+        start_date = pull_last_start_date(ti)
+        if not start_date:
+            first_release = True
+            start_date = pendulum.instance(kwargs['dag'].default_args['start_date']).start_of('day')
+        start_date = start_date - timedelta(days=1)
+        end_date = pendulum.utcnow()
+        logging.info(f'Start date: {start_date}, end date: {end_date}, first release: {first_release}')
+
+        release = self.create_release(start_date, end_date, first_release)
+
+        ti.xcom_push(self.RELEASES_XCOM, release)
+        ti.xcom_push(self.LAST_START_XCOM, end_date)
+
+    @abstractmethod
+    def download(self, release):
+        """
+        Download code.
+        :param release:
+        """
+        raise NotImplementedError
+
+    # method that calls an abstractmethod that needs to be customised by the user
+    def download_task(self, **kwargs) -> bool:
+        """ Download release data to disk.
+
+        :param kwargs: The context passed from the PythonOperator.
+        :return: None.
+        """
+        ti: TaskInstance = kwargs['ti']
+        release = pull_release(ti)
+
+        success = self.download(release)
+        return True if success else False
```

Comment:
This method should override an abstract method from the `Telescope` class rather than calling an abstract method from the `Telescope` class.